src/script/miniscript.h

```diff
@@ -544,17 +562,23 @@ struct Node {
         return TreeEval<NodeRef<Key>>(upfn);
     }
 
+    Fragment GetFragment() const { return fragment; }
+    uint32_t GetK() const { return k; }
+    const std::vector<Key>& GetKeys() const { return keys; }
```

Comment:
do we need to return vectors here? Seems overly specific, I think spans should also suffice here